pyanaconda/modules/network/config_file.py

```diff
@@ -44,10 +44,14 @@ def get_config_files_paths(root_path=""):
     paths = []
     for directory in (os.path.normpath(root_path + IFCFG_DIR),
                       os.path.normpath(root_path + KEYFILE_DIR)):
-        for filename in os.listdir(directory):
-            if not (_has_keyfile_basename(filename) or _has_ifcfg_basename(filename)):
-                continue
-            paths.append(os.path.join(directory, filename))
+        try:
+            for filename in os.listdir(directory):
+                if not (_has_keyfile_basename(filename) or _has_ifcfg_basename(filename)):
+                    continue
+                paths.append(os.path.join(directory, filename))
+        except FileNotFoundError:
+            # ignore missing directories
```

Comment:
Could you please add debug info. It could help with the debugging in the future.

Reply:
Maybe just checking directory existence instead of catching an exception would be more self-explaining here?

Reply:
"easier to ask for forgiveness than permission" is more pythonic than "look before you leap" -- and the latter is a bad style to get used to, as it is TOCTOU race condition. Sure, in this case it isn't relevant (most probably), but easier/more robust to just handle the exception.

Reply:
I agree with the general rule, in this case I was considering the asking because
- the race condition would not apply
- the readability
- the block that is caught seemed too big to me (compared to the os.listdir call)
But I am fine with catching the exception either.

Reply:
I would agree, but in this case, it is not very clear which file might be missing since we are working with directories and files in these directories. Also, the robustness is not always helpful. Anaconda sometimes runs in very weird and unstable environments (for example, with failing hardware) and it doesn't make sense to pretend that everything is fine when it clearly isn't.

Reply:
OK, changed to LBYL now.